hs_core/hydroshare/resource.py

```diff
@@ -559,13 +559,20 @@ def create_new_version_resource(ori_res, new_res):
         res_avail_date = new_res.metadata.dates.all().filter(type='available')[0]
         new_res.metadata.create_element('date', type='available', start_date=res_avail_date.start_date, end_date=res_avail_date.end_date)
 
-    # add Relation element to link source and target resources
+    # add or update Relation element to link source and target resources
     if new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier"):
         hs_identifier = new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
         ori_res.metadata.create_element('relation', type='isReplacedBy', value=hs_identifier.url)
     else:
         ori_res.metadata.create_element('relation', type='isReplacedBy', value=new_res.short_id)
```

Comment:
I hate to be adjusting policy in a line by line review of a pull request.  I would prefer this be addressed holistically in terms of the publications workflow and specifications document.  It would be worth going back and looking at that to see what it says.  I think we need the following:

- An owner can publish a new version of a published resource, so this specific relationship metadata of the published resource can be changed.   
- I think that a relationship named for "superceded" rather than "replaced" may be better, because the published resource is still a matter of record.
- The time that a new version is created is not the right time to add the superceded or replaced relationship because the workflow will most likely be to (1) make a new version, then do some work on it that eventually (perhaps months or years later) results in it being ready to supercede the initially published resource.  Only when the new superceding resource is ready, perhaps as indicated by it being "published" should the relationship be added to the old published resource.  Until then the published version is the version of record.  

We need to figure out simple tools and guidance for users to get this right.